### DISCUSSION TITLE
CRITICAL FAILURE: Camera sensor failed to initialise. A full (hard, power off/on) reboot will probably be needed to recover from this. Meanwhile; this unit will reboot in 1 minute since these errors sometime clear automatically

### DISCUSSION THREAD

**Discussion by Tecnhotron** [easytarget/esp32-cam-webserver#207] [Common Issues] [answered]
Hello everyone, I have ESP-32 Cam model Ai-THINKER (clone) with esp32-cam-webserver code uploaded. Till yesterday it was working fine but when i was moving the camera it suddenly stopped working and gave me this error:.

ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:1216
ho 0 tail 12 room 4
load:0x40078000,len:10944
load:0x40080400,len:6360
entry 0x400806b4

====
esp32-cam-webserver: ESP32 camera server
Code Built: Jan  2 2022 @ 21:18:27
Base Release: 4.0.beta2
[E][camera.c:1113] camera_probe(): Detected camera not supported.
[E][camera.c:1379] esp_camera_init(): Camera probe failed with error 0x20004

CRITICAL FAILURE: Camera sensor failed to initialise.

A full (hard, power off/on) reboot will probably be needed to recover from this.
Meanwhile; this unit will reboot in 1 minute since these errors sometime clear automatically
Lamp: 0%, pwm = 0
Starting WiFi
Known external SSIDs: None
MAC address: 94:B9:7E:E6:0D:1C
AccessPoint mode selected in config
Setting up AccessPoint
  SSID     : ESP32-CAM-CONNECT
  Password : InsecurePassword
IP address: 192.168.4.1
Setting httpURL
Starting Captive Portal
Setting up OTA

No OTA password has been set! (insecure)

Time functions disabled
Starting web server on port: '80'
Starting stream server on port: '81'

Camera unavailable due to initialisation errors.

This is the 4.0 alpha
 - Face detection has been removed!

![image](https://user-images.githubusercontent.com/69301779/147881323-27fa8274-df20-40d9-80bf-9696ac4afb70.png)

**Comment by founfabug** [comment]
As nothing else changed, you can try re-seating the camera connector. It may have come out of alignment with the move. Videos available online, just be gentle.

**Comment by Tecnhotron** [comment]
I removed the camera sensor 10 times but no change, then I reuploaded the but same

**Comment by bolulu** [comment]
I got the same picture, with Ai-thinker CAM
![image](https://user-images.githubusercontent.com/24476168/148248433-2e449f9a-87ac-4813-9bb8-1e2bf045d8ae.png)

**Comment by easytarget** [comment, marked as answer]
### I'm moving this to a discussion since It is a good issue to illustrate this common problem

There are numerous ways these failures can appear. 
- With a bit of googling you will find that initialisation errors are very, very common with these modules.
- There are both hardware, and software related reasons for this.
- None of them are specific to my version of this sketch (which uses the same initialisation+setup as the official example)

### Typical causes:
- Low power/voltage to the unit
  - this is the number 1 cause, and can strike randomly
  - cheap clones are extremely sensitive to this.
  - Also see the next point (XCLK) since this can help overcome this issue.
  - leaving the unit unplugged for a while can help. I think it helps if the unit is not warm when initialising
- `XCLK` frequency set too high.
  -  This setting (which can be found in the `myconfig.h` settings, and also in the GUI, controls the speed of the bus that is used to communicate between the ESP32 abd the camera module. If this is failing you can get errors like to ones above.
  - I have encountered examples of cheap clone modules where this setting needs to be `2MHz` in place of the default `8MHz`
- Broken camera module on the board; the cables and connectors are fragile and easy to damage.
- Frankly; some of the clones out there are very badly made.
  - I have several units, some are better than others, the ones I brought recently (late 2021) seem particularly cheap.
  - [rant] What happens is that manufacturers buy up 'out of spec' reject camera modules and esp moodules  from the back of the supply chain, then build modules using these and slip them out onto the market at a lowish price. There is nothing unusual in this, it has always happened, ever brought something 'cheap' that turned out to actually be 'defective', well that's capitalist economics working as intended, caveat emptor. And no, it's not just confined to China.
- Bad arduino core; running with a corrupted or outdated ESP arduino core in your build system will not help.

### Yes; I see errors like this occasionally
I understand that they are frustrating and a bit random, you have my sympathies and understanding; but I cannot provide any quick fixes.
- This is why I added the error catching, reporting and rebooting functionality to the sketch.
- _It always goes away after a while without me actually 'fixing' anything_.
  - The only board I had that didn't recover was because I had twisted the camera module on a board accidentally (broken ribbon cable). Fitting a new camera module fixed it, I have some wide-angle modules and spares.

Edited by @easytarget , 25/5/2022, added `XCLK` note

**Reply by Tecnhotron** [reply]
I think "I twisted the camera module on a board accidentally" so I'll buy another one

**Comment by DSerejo** [comment]
Not sure this will help anyone else.
I was having this same problem and got it to work by using this board config below

![board_config](https://user-images.githubusercontent.com/10634013/150619742-8f493bc3-a529-45c5-86b1-82eb8c4beee1.PNG)

**Comment by Tecnhotron** [comment]
It was a physical camera problem
![Uploading 20220122_195835~2.jpg…]()

**Comment by badyast** [comment]
I solved it setting a XCLK_FREQ at the bottom of myconfig.h. I tried both examples (20000000 and 3000000).
